### DISCUSSION TITLE
Cannot interact with a custom build in KVM Virt-Manager

### DISCUSSION THREAD

**Discussion by manujchandra** [SpiralLinux/SpiralLinux-project#422] [Q&A] [answered]
Greetings,

Here are the steps I am following:

- I downloaded SpiralLinux_XFCE_12.231120_x86-64.iso from Sourceforge and created a VM in VirtManager (Virtual Machine Manager/KVM) and it works fine.

- Then I downloaded [Spiral XFCE](https://github.com/SpiralLinux/SpiralLinux-project/blob/main/SpiralLinux-XFCE.tar.gz) and unzipped it and **without making any modifications**, made an ISO

- I took the generated ISO and created a new VM. It boots into the desktop, but keyboard and mouse are not working, I cannot click on anything or even send keystrokes using menu. It's just like a picture of the desktop and no keystroke is working and nothing can be clicked.

![Screenshot_2024-04-22_10-39-18](https://github.com/SpiralLinux/SpiralLinux-project/assets/15191493/5b5dcbff-9ef0-420a-906d-4dbb446cd4b3)

- If I force shutdown and start again, it never runs at all. I delete the VM.

- Next, I made 3 modifications, added mpv from apt, KeepPassXC using flatpak and VSCodium using a .deb

- I made another build, and it would also not run in the VM. (Just a reminder, the Sourceforge Spiral Linux ISO runs fine in the VM)

- I burned the custom ISO on a USB and booted into a real laptop. It seems to be working fine and mpv, KeepassXC and VSCodim can be executed.

Any idea why the custom ISO is not working in the KVM? (I am using Spiral Linux 12 and the builds are also made using the same). I want to experiment making custom ISOs and install them in KVM for testing as it would be faster.

[build.log](https://github.com/SpiralLinux/SpiralLinux-project/files/15057925/build.log)

Build logs are attached above.

I also observed a warning, don't know if it's related:

`Warning: Failed to install org.freedesktop.Platform.openh264: While trying to apply extra data: apply_extra script failed, exit status 256
`

Thanks for the help. 🙏🏽

**Comment by manujchandra** [comment]
PS : KVM can be installed in Sprial Linux by installing the following packages:

`qemu-kvm
qemu-system
qemu-utils
python3
python3-pip
libvirt-clients
libvirt-daemon-system
bridge-utils
virtinst
libvirt-daemon
virt-manager
qemu-system-gui
gir1.2-spiceclientgtk-3.0`

**Comment by geckolinux** [comment, marked as answer]
> Any idea why the custom ISO is not working in the KVM? (I am using Spiral Linux 12 and the builds are also made using the same).

Hi there, yes, this is a known bug:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1051547
You need to build your images with a Debian 11 Bullseye system, and the SpiralLinux build recipes have some tweaks to work around the issues caused by Debian 11 building a Debian 12 system.

Another alternative would be to disable the Backports repo, as the issue only happens with the kernel from Backports.

**Comment by manujchandra** [comment]
Hello @geckolinux ,

Thanks for your answer. I disabled the backports by setting `--backports false` in `SpiralLinux-XFCE/auto/iso` and deleting the backports file in `SpiralLinux-XFCE/config/includes.chroot/etc/apt/preferences.d`

Hopefully this is the correct way of disabling back ports? 

I was able to run the ISO in KVM. Thanks.

**Reply by geckolinux** [reply]
Hi there, yes, that sounds right.
